src/launcher/java/org/truffleruby/launcher/RubyLauncher.java

```diff
@@ -59,6 +59,9 @@ protected void printVersion() {
 
     @Override
     protected List<String> preprocessArguments(List<String> args, Map<String, String> polyglotOptions) {
+        // This is the last place we can set a system property before the Ruby language is created
+        System.setProperty("truffleruby.single_threaded", Boolean.FALSE.toString());
```

Comment:
Note: Before the Main class is removed it might have to be set in there too (depending on the merging order). They share `Launcher` class not the `RubyLauncher` class.